src/Microsoft.Extensions.Configuration.AzureAppConfiguration/ConfigurationSettingPageableManager.cs

```diff
@@ -0,0 +1,19 @@
+﻿using Azure.Data.AppConfiguration;
+using Azure;
+using System.Collections.Generic;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration
+{
+    internal class ConfigurationSettingPageableManager
+    {
+        public virtual IAsyncEnumerable<Page<ConfigurationSetting>> GetPages(AsyncPageable<ConfigurationSetting> pageable, IEnumerable<MatchConditions> matchConditions)
```

Comment:
Why are these methods virtual?

Reply:
Thanks, I didn't document this anywhere so good callout. The approach recommended by the SDK team for testing the `AsPages` extension with `MatchConditions` as shown here was to create a virtual method that we could override in the tests using the `MockConfigurationSettingPageableManager` class. Because `Moq` doesn't let you mock extension methods, this was the only approach that we came up with. Any details I missed are probably in the issue I linked in the PR description.